src/io.c

```diff
@@ -379,6 +383,47 @@ static void __fs_flush(void)
     }
 }
 
+#define SYNC_FILE_RANGE
+
+#ifdef SYNC_FILE_RANGE
+#define CHUNK_SIZE	(512 * 1024 * 1024)
+/* NOTE: Using sync_file_range() function does not portable */
+void fs_sync(void)
+{
+    off_t offset = 0;
+    size_t chunk_size = CHUNK_SIZE;
+    size_t bytes_to_sync = 0;
+
+    /* request writeback */
+	while (offset < dev_size) {
+		bytes_to_sync = (dev_size - offset) < chunk_size ?
+			(dev_size - offset) : chunk_size;
+
+		// Optionally, you can also use sync_file_range to wait for the write to complete
+		if (sync_file_range(fd, offset, dev_size,
```

Comment:
I wonder what this loop means. if calling with SYNC_FILE_RANGE_WAIT_BEFORE, sync_file_range waits for the previous call of sync_file_range.

Reply:
modified dev_size -> bytes_to_sync

Reply:
Could you elaborate why SYNC_FILE_RANGE_WAIT_BEFORE is required?

Reply:
> Could you elaborate why SYNC_FILE_RANGE_WAIT_BEFORE is required?

I refered sync_file_range() man page
`       SYNC_FILE_RANGE_WAIT_BEFORE | SYNC_FILE_RANGE_WRITE
              Ensures   that   all   pages  in  the  specified  range  which  were  dirty  when
              sync_file_range() was called are placed under write-out.  This is a  start-write-
              for-data-integrity operation.

       SYNC_FILE_RANGE_WRITE
              Start write-out of all dirty pages in the specified range which are not presently
              under write-out.  This is an asynchronous flush-to-disk operation.  This  is  not
              suitable for data integrity operations.
`